mtc.go

```diff
@@ -899,6 +924,80 @@ func (a *AbridgedAssertion) unmarshal(s *cryptobyte.String) error {
 	return nil
 }
 
+func (a *AssertionRequest) UnmarshalBinary(data []byte) error {
+	var (
+		s cryptobyte.String = cryptobyte.String(data)
+	)
+	a.Checksum = make([]byte, sha256.Size)
+	if !s.CopyBytes(a.Checksum) {
+		return ErrTruncated
+	}
+
+	err := a.Assertion.unmarshal(&s)
```

Comment:
The main reason `AssertionRequest` lives in `mtc` instead of the `ca` package is because it's convenient to call the `unmarshal` function here. Otherwise, we have to add length prefixes to the `Assertion` and `Evidence` to be able to use the `UnmarshalBinary` functions.

Reply:
We could make `unmarshal` a Public function in an internal package if we wanted to hide AssertionRequest, but let's not bother about that now.